components/scream/src/interface/scream_scorpio_interface.F90

```diff
@@ -0,0 +1,933 @@
+  type hist_var_list
+    type(hist_var_t),    pointer :: var => NULL()  ! Pointer to a history variable structure
+    type(hist_var_list), pointer :: next => NULL() ! Needed for recursive definition
+  end type hist_var_list
+!----------------------------------------------------------------------
+  type pio_file_list
+    type(pio_atm_output), pointer :: pio_file => NULL() ! Pointer to an atm. pio file
+    type(pio_file_list),  pointer :: next => NULL()     ! Needed for recursive definition
+  end type pio_file_list
+  ! Define the first pio_file_list
+  type(pio_file_list), target :: pio_file_list_top
+!----------------------------------------------------------------------
+  type, public :: pio_atm_output
+        !> @brief Output filename.
+        character(len=max_chars) :: filename = ""
+
+        !> @brief Contains data identifying the file.
+        type(file_desc_t)     :: pioFileDesc
+
+        !> @brief Number of output dimensions, and counter to track them during
+        !  registration
+        integer               :: DimCounter = 0
+        !> @brief Recursive list of variables
+        type(hist_coord_list)   :: coord_list_top
+
+        !> @brief Number of output variable and counter to track them during
+        !  registration
+        integer               :: VarCounter = 0
+        !> @brief Recursive list of variables
+        type(hist_var_list)   :: var_list_top
+
+        !> @brief Number of history records on this file
+        integer               :: numRecs
+
+        !> @brief Coordinate Dimensions Array
+        type(hist_coord_t), allocatable :: dimensions(:)
+
+  end type pio_atm_output
+
+!----------------------------------------------------------------------
+  interface grid_write_data_array
+    module procedure grid_write_darray_1d_int
+    module procedure grid_write_darray_2d_int
+    module procedure grid_write_darray_3d_int
+    module procedure grid_write_darray_1d_double
+    module procedure grid_write_darray_2d_double
+    module procedure grid_write_darray_3d_double
+    module procedure grid_write_darray_1d_real
+    module procedure grid_write_darray_2d_real
+    module procedure grid_write_darray_3d_real
+  end interface
+!----------------------------------------------------------------------
+contains
+!=====================================================================!
+  subroutine eam_init_pio_1(mpicom,atm_id)
+
+    integer, intent(in) :: mpicom  ! MPI communication group for ATM
+    integer, intent(in) :: atm_id  ! Unique identifier assigned by component coupler
+
+    type(pio_atm_output), pointer :: current_atm_file => null()
+    integer :: ierr
+    logical :: found
+
+    ! Gather the pio_subsystem information for the atmosphere component.
+    call eam_init_pio_subsystem(mpicom,atm_id)
+
+    ! Allocate the dimension and variable arrays for the output file.
+    call get_pio_atm_file("example_pio_structured.nc",current_atm_file,found)
```

Comment:
No, these are just placeholders.  I want to have a solid file to test against when I begin phase 2 with the C++ development.  Once that is set up then I won't have any need for these files and can use proper unit tests to check future development.

Reply:
I know you know that they are placeholders, and you are 99.99% likely to work on phase 2. BUT, just in case something comes up and someone else has to take over the scorpio work, please mark clearly _all_ the things that are only placeholder for this initial stage, and will have to be removed in the final product.

Reply:
I've added much more robust comments.